revolv/project/tests/test_integration.py

```diff
@@ -0,0 +1,185 @@
+
+    def test_admin_form_submission(self):
+        """
+        Tests that administrators can create project updates.
+        """
+        self.test_profile.make_administrator()
+        self.helper_test_create_update_ambassador_administrator()
+
+    def test_ambassador_form_submission(self):
+        """
+        Tests that ambassadors can create project updates.
+        """
+        self.test_profile.make_ambassador()
+        self.helper_test_create_update_ambassador_administrator()
+
+    def test_donor_form_submission(self):
+        """
+        Tests that donors can't create project updates.
+        """
+        self.test_profile.make_donor()
+        response = self.send_test_user_login_request()
+        self.assertUserAuthed(response)
+        project = Project.factories.base.create()
+        response = self.client.post('/project/%d/update' % project.pk, {'update_text': 'This is an update'})
+        
+        #the deny_access method is called in the view from the userdatamixin, which returns a redirect to the homepage
+        self.assertEqual(response.status_code, 302)
+        
+class EditProjectUpdatesTest(TestUserMixin, UserTestingMixin, TestCase):
+
+    def _make_update(self, project, text):
+        """
+        Makes an update to of given text to the given project.
+        """
+        response = self.send_test_user_login_request()
+        self.assertUserAuthed(response)
+        #response = self.client.post('/project/%d/update' % project.pk, {'update_text': text})
+        ProjectUpdate(project = project, update_text = text).save()
+        new_updates = ProjectUpdate.objects.filter(update_text = text)
+        return new_updates[0]
+
+    def _helper_test_edit_update(self, donor = False):
+        """
+        Tests that the administrator and ambassador can make updates. 
+        """
+        project = Project.factories.base.create()
+        update = self._make_update(project, "This update has not been changed.")
+
+        response = self.client.post('/project/editupdate/%d' % update.pk,
+                        {'update_text': 'This update has been changed.'})
+
+        old_update_text_updates = ProjectUpdate.objects.filter(update_text="This update has not been changed.")
+        new_updates = ProjectUpdate.objects.filter(update_text="This update has been changed.")
+
+        if not donor:
+            self.assertEqual(len(old_update_text_updates), 0)
+            self.assertEqual(update.project, new_updates[0].project)
+            self.assertEqual(new_updates[0].update_text, "This update has been changed.")
+        else:
+            #the donor should be denied access and redirected
+            self.assertEqual(response.status_code, 302)
+
+    def test_edit_update_administrator(self):
+        """
+        Tests that a change is made to an update for an administrator.
+        """
+        self.test_profile.make_administrator()
+        self._helper_test_edit_update()
```

Comment:
Might as well pass in the False kwarg here for readability purposes, I think. `donor=False` is a lot more descriptive than just nothing